integreat_cms/api/decorators.py

```diff
@@ -117,3 +125,74 @@ def wrap(request, *args, **kwargs):
             return JsonResponse({"error": str(e) or "Not found."}, status=404)
 
     return wrap
+
+
+def matomo_tracking(func):
+    """
+    This decorator is supposed to be applied to API content endpoints. It will track
+    the request in Matomo. The request to the Matomo API is executed asynchronously in its
+    own thread to not block the Integreat CMS API request.
+
+    Only the URL and the User Agent will be sent to Matomo.
+
+    :param func: decorated function
+    :type func: ~collections.abc.Callable
+
+    :return: The decorated feedback view function
+    :rtype: ~collections.abc.Callable
+    """
+
+    @wraps(func)
+    def wrap(request, *args, **kwargs):
+        r"""
+        The inner function for this decorator.
+
+        :param request: Django request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied kwargs
+        :type \**kwargs: dict
+
+        :return: The response of the given function or an 404 :class:`~django.http.JsonResponse`
+        :rtype: ~django.http.JsonResponse
+        """
+        data = {
+            "idsite": request.region.matomo_id,
+            "token_auth": request.region.matomo_token,
+            "rec": 1,
+            "url": request.build_absolute_uri(),
+            "urlref": settings.BASE_URL,
+            "ua": request.META["HTTP_USER_AGENT"]
+            if "HTTP_USER_AGENT" in request.META
+            else "unknown user agent",
```

Comment:
```suggestion
            "ua": request.META.get("HTTP_USER_AGENT", "unknown user agent"),
```
I think using a default dict value here is more readable.

Reply:
Indeed more readable.